src/background.ts

```diff
@@ -151,14 +166,330 @@ const onStartUp = async () => {
   checkIfProtected(store.getState());
 
   browser.tabs.onUpdated.addListener(TabEvents.onDomainChange);
+  browser.tabs.onUpdated.addListener(TabEvents.onTabDiscarded);
   browser.tabs.onUpdated.addListener(TabEvents.onTabUpdate);
   browser.tabs.onRemoved.addListener(TabEvents.onDomainChangeRemove);
-  browser.tabs.onRemoved.addListener(TabEvents.cleanFromFromTabEvents);
+  browser.tabs.onRemoved.addListener(TabEvents.cleanFromTabEvents);
 
   // This should update the cookie badge count when cookies are changed.
   browser.cookies.onChanged.addListener(CookieEvents.onCookieChanged);
+
+  if (browser.contextMenus) {
+    ContextMenuEvents.menuInit(store.getState());
+    if (!browser.contextMenus.onClicked.hasListener(onContextMenuClicked)) {
+      browser.contextMenus.onClicked.addListener(onContextMenuClicked);
+    }
+  }
+  browser.browserAction.setTitle({ title: `${mf.name} ${mf.version} [READY] (0)` });
+  if (currentSettings.debugMode.value) {
+    cadLog({
+      msg: `background.onStartUp is complete.`,
+      type: "info"
+    });
+  }
 };
 
+async function onContextMenuClicked(
```

Comment:
There's a lot of code here, can we move this function to another file?

Reply:
I'll move it into ContextMenuEvents.

Reply:
Should be resolved through 2b5e3b2